frontend/src/general/dialog/__tests__/DeleteStreamDialog.test.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import Enzyme from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import { beforeEach, describe, expect, jest } from "@jest/globals";
+import DeleteStreamDialog from "../DeleteStreamDialog";
+import Dialog from "../Dialog";
+
+import * as StreamApi from "../../../api/StreamApi";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("<DeleteStreamDialog/> class", () => {
+  let wrapper;
+
+  const mockPush = jest.fn();
+  const mockGo = jest.fn();
+  const dummyHistory = {
+    push: mockPush,
+    go: mockGo
+  };
+  const mockOpenDialog = jest.fn();
+  const mockCloseDialog = jest.fn();
+  const mockRefElement = {
+    current: {
+      openDialog: mockOpenDialog,
+      closeDialog: mockCloseDialog
+    }
+  };
+  const dummyId = "dummyId";
+  beforeEach(() => {
+    jest.spyOn(React, "createRef").mockImplementation(() => {
+      return mockRefElement;
+    });
+    jest.spyOn(StreamApi, "deleteStream").mockImplementation(() => {
+      return Promise.resolve();
+    });
+    wrapper = Enzyme.shallow(
+      <DeleteStreamDialog deleteId={dummyId} history={dummyHistory} />
+    );
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("render() function", () => {
+    it("renders one <Dialog/> Component", () => {
+      expect(wrapper.find(Dialog)).toHaveLength(1);
+    });
+  });
+  describe("openDialog() function", () => {
+    it("calls the child's openDialog() function", () => {
+      wrapper.instance().openDialog();
+      expect(mockOpenDialog).toBeCalledTimes(1);
+    });
+  });
+  describe("confirmDelete() function", () => {
+    it("calls StreamAPI with the passed ID", () => {
+      wrapper.instance().confirmDelete();
+      expect(StreamApi.deleteStream).toBeCalledWith(dummyId);
+    });
+  });
+  describe("afterDelete() function", () => {
+    it("calls the correct functions", () => {
```

Comment:
Rename to describe better what it calls (e.g. Closes the dialog and refreshes the page)

Reply:
fixed in 7af4db61